### PR TITLE
feat(cli): add basic tests

### PR DESCRIPTION
**Pull request by akshatdubeysf**
BREAKING CHANGE:
moved prompts to commands

## Description

- Refactored to prevent duplication of prompt code.
- Add basic tests for CLI commands.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [X] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -22,12 +22,12 @@
     "copydeps": "copyfiles -a --up 1 src/**/templates/** lib ",
     "lint": "eslint ./src --ext .js,.ts && ejslint ./src",
     "posttest": "npm run lint",
-    "prepack": "npm run build",
-    "version": "oclif-dev readme && git add README.md",
-    "tsc": "tsc -b",
     "build": "rimraf lib && tsc -b && oclif-dev manifest && oclif-dev readme && npm run copydeps",
-    "postbuild": "rimraf oclif.manifest.json",
-    "test": "npm run build"
+    "version": "oclif-dev readme && git add README.md",
```
Comment (akshatdubeysf) [inline packages/cli/package.json:26]: restore postbuild function to remove manifest

File: packages/cli/src/__tests__/helper/command-test.helper.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import {Config} from '@oclif/config';
+import {expect} from 'chai';
+import {createStubInstance, stub} from 'sinon';
+import Environment from 'yeoman-environment';
+import {AnyObject, CommandTestCase} from '../../types';
+
+export function commandTest(
+  testCase: CommandTestCase,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  command: new (...args: any[]) => Command,
```
Comment (akshatdubeysf) [inline packages/cli/src/__tests__/helper/command-test.helper.ts:11]: Create a type for this

File: packages/cli/package.json (hunk 1 above)
Comment (AnkurBansalSF) [reply packages/cli/package.json:25]: Done

File: packages/cli/src/__tests__/helper/command-test.helper.ts (hunk 2 above)
Comment (AnkurBansalSF) [reply packages/cli/src/__tests__/helper/command-test.helper.ts]: Done

**Pull request merged**